### PR TITLE
Docs: Document Prometheus-Alertmanager integration settings

### LINKED ISSUE #46908: Docs: Missing option in datasource api to control alerting section
**What happened**:

Would like to provision datasource from json as in [this documentation](https://github.com/grafana/grafana/blob/main/docs/sources/datasources/prometheus.md) to setup alerting section

**What you expected to happen**:

An option to set Alerting section

image

**How to reproduce it (as minimally and precisely as possible)**:

Upload datasource yaml file during deployment

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.4.4
- Data source type & version: prometheus and loki
- OS Grafana is installed on: Ubuntu 20.04
- User OS & Browser: Ubuntu 20.04, Chrome
- Grafana plugins: N/A
- Others: N/A

### LINKED ISSUE #53263: Docs feedback: /datasources/alertmanager.md
I followed the yaml example [on this page](https://grafana.com/docs/grafana/latest/datasources/alertmanager/) to add an alertmanager datasource, but it threw errors in the UI.

After searching through the forums, I was able to get it working by adding the implementation to the datasource:

```
    jsonData:
      implementation: prometheus
```

As far as I can tell, this implementation field isn't documented anywhere, nor are its values. I only found out about it by reading a commenter on [this issue](https://github.com/grafana/grafana/issues/50648), who in turn found it on a blog.

### PR DESCRIPTION
**Pull request by gguillotte-grafana**
**What this PR does / why we need it**:

Documents `alertmanagerUid` and `manageAlerts` for the Prometheus data source, and `implementation` for the Alertmanager data source.
These were implemented without documentation around Grafana v8.3.

**Which issue(s) this PR fixes**:

Fixes #46908.
Fixes #53263.

Documents the cause of #50648.

**Special notes for your reviewer**:

This does not document how a user is expected to find the `alertmanagerUid` during provisioning. It's possible to hardcode the `uid` in the Alertmanager data source when provisioning it, then use that UID as the `alertmanagerUid`, but it's unclear whether that's advisable. See #46908 for a discussion of this problem.

### CODE REVIEW

**Comment by gguillotte-grafana** [comment]
Moving out of draft to request a review. I would prefer to resolve the question about `alertmanagerUid` best practices before merging this, but wouldn't block on it.

File: docs/sources/datasources/prometheus.md
Comment (konrad147) [inline docs/sources/datasources/prometheus.md:34]: We're going to remove this option because there is no way to check if it aligns with the actual Prometheus configuration https://github.com/grafana/grafana/pull/57369

**Comment by armandgrillet** [comment]
@brendamuir What should we do with that PR? I see that the branch has conflicts with `main`.

**Review by brendamuir — APPROVED** [review]

File: docs/sources/datasources/prometheus.md
Comment (konrad147) [inline docs/sources/datasources/prometheus.md:291]: Should be removed. We no longer support this option

File: docs/sources/datasources/alertmanager/_index.md
Comment (brendamuir) [inline docs/sources/datasources/alertmanager/_index.md:28]:
```suggestion
When using Prometheus, contact points and notification policies are read-only in the Grafana Alerting UI, because it doesn't support updates to the configuration using HTTP API.
```

**Pull request merged**
